ironfish/src/network/peers/addressManager.ts

```diff
@@ -12,71 +13,113 @@ import { PeerManager } from './peerManager'
  * and provides functionality for persistence of said data.
  */
 export class AddressManager {
+  LIMIT = 50
+
   hostsStore: HostsStore
   peerManager: PeerManager
+  peerIdentityMap: Map<Identity, PeerAddress>
 
   constructor(hostsStore: HostsStore, peerManager: PeerManager) {
     this.hostsStore = hostsStore
     this.peerManager = peerManager
-  }
+    // load prior peers from disk
+    this.peerIdentityMap = new Map<string, PeerAddress>()
+    const currentTime = Date.now()
+    let priorPeers = this.hostsStore.getArray('priorPeers')
 
-  get priorConnectedPeerAddresses(): ReadonlyArray<Readonly<PeerAddress>> {
-    return this.hostsStore.getArray('priorPeers')
-  }
-
-  /**
-   * Returns a peer address for a disconnected peer by using current peers to
-   * filter out peer addresses. It attempts to find a previously-connected
-   * peer address that is not part of an active connection.
-   */
-  getRandomDisconnectedPeerAddress(peerIdentities: string[]): PeerAddress | null {
-    if (this.priorConnectedPeerAddresses.length === 0) {
-      return null
+    // If there are more than 50 peers, we remove
+    // extra peers from the list. This should only happen during
+    // the first time the node is started after this change is implemented.
+    if (priorPeers.length > this.LIMIT) {
+      priorPeers = priorPeers.slice(0, this.LIMIT)
     }
 
-    const currentPeerIdentities = new Set(peerIdentities)
+    for (const peer of priorPeers) {
```

Comment:
should we filter for peers that don't have identity, address or port before we slice with the limit. That way we don't slice out all of the good options

Reply:
moved the filter to after this loop